File: Programming/B04_Java/S1_Basics/S1_05_Control_flow.java
/*
    Control flow
*/

//T// Contents
//T// Decision making
//T// Loops

public class Section1_8
{
    public static void main(String args[])
    {

//T// Decision making

        int i1 = 11;
//T// if, else if, else statement
        if (i1 < 10)
        {
            System.out.println("Value is less than ten");
        }else if (i1 == 10)
        {
            System.out.println("Value is ten");
        }else
        {
            System.out.println("Value is greater than ten");
        }

//T// switch, case, default statement
        switch (i1)
        {
            case 1:
                System.out.println("Value 1");
                break;
            case 11:
                System.out.println("Value 11");
                break;
            default:
                System.out.println("Other value");
        }

//T// Loops

int i1 = 5;
//T// while loop
        while (true)
        {
            System.out.println("Infinite while loop?");
            i1++;
            if (i1 == 7) break;
        }

//T// for loop
        for (i1=i1;i1<=10;i1++)
        {
            if (i1 == 8) continue;
            System.out.println("In for loop");
        }

//T// do while loop
        do
        {
            System.out.println("Do while loop");
        } while (i1 < 10);

        int[] arr1 = {6,2,38,3};
//T// for loop in array
        for(int i2:arr1)
        {
            System.out.println("Current value: "+i2);
        }
    }
}

File: Programming/C04_PHP/S1_Basics/S1_05_Control_flow.php
<!--
    Control flow
-->

<html>
    <head><title>Control flow</title></head>
    <body>
    <?php

//T// Contents
//T// Decision making
//T// Loops

//T// Decision making

    $a = 1;
    echo "a value is: $a<br/>";
//T// if, elseif, else statement
    if ($a > 0)
    {
        echo "a is greater than 0<br/><br/>";
    }
    elseif ($a == 0)
    {
        echo "a equals 0<br/><br/>";
    }
    else
    {
        echo "a is less than 0<br/><br/>";
    }

//T// switch statement
    switch ($a)
    {   
        case 0:
            echo "a is the number zero<br/>";
        case "str1":
            echo "a is a string.. or is it?<br/>";
            break;
        case 1:
            echo "a is the number one<br/>";
            break;
        default:
            echo "a wasn't matched to any value<br/>";
    }

//T// Loops

//T// for loop, continue
    for ($i = 2;$i < 6;$i++)
    {
        if ($i == 3) continue;
        echo "i is $i<br/>";
    }

    echo "<br/>";
//T// while loop, nested loop
    while ($i > 4)
    {
        for ($j = 0;$j < 2;$j++)
        {
            echo "i is $i, j is $j<br/>";
        }
        $i--;
    }

    echo "<br/>";
//T// do while loop, break
    do
    {
        echo "i is $i<br/>";
        $i++;
        if ($i == 8) break;
    } while ($i < 50);

    echo "<br/>";
    $arr1 = array(12,"str1",7);
//T// foreach as loop
    foreach ($arr1 as $valEl)
    {
        echo "Current value un array is $valEl<br/>";
    }

    ?>
    </body>
</html>